### PR TITLE
Feature gw redundancy

### PR DESCRIPTION
**Pull request by Rahul-D78**
This PR introduces multiple replicas of the VPN pods, multiple network connections to the remote cluster, with ECMP support to handle failure scenarios effectively.

### CODE REVIEW

**Comment by saivenkatesh1909** [comment]
report link  'https://kubeslice.github.io/e2e-allure-reports/Kind-worker-operator-2022-10-16T18:36:14-master-174/index.html'

**Comment by saivenkatesh1909** [comment]
report link  'https://kubeslice.github.io/e2e-allure-reports/Kind-worker-operator-2022-10-18T07:41:00-master-175/index.html'

File: .github/workflows/scripts/start-kind.sh [hunk 1]
```diff
@@ -47,8 +47,10 @@ WorkerChartOptions:
     "operator.image": "worker-operator"
     "operator.tag": "${GITHUB_HEAD_COMMIT}"
 TestSuitesEnabled:
+  EmptySuite: true
   HubSuite: false
   WorkerSuite: true
+  IstioSuite: false
```
Comment (kmjayadeep) [inline .github/workflows/scripts/start-kind.sh:53]: please enable this before merging the PR

File: api/v1beta1/slicegateway_types.go [hunk 2]
```diff
@@ -53,6 +53,8 @@ type SliceGatewayConfig struct {
 	SliceGatewayNodePort int `json:"sliceGatewayNodePort,omitempty"`
 	// Remote Node IP
 	SliceGatewayRemoteNodeIP string `json:"sliceGatewayRemoteNodeIp,omitempty"`
+	// Remote Node IPs
```
Comment (kmjayadeep) [inline api/v1beta1/slicegateway_types.go:56]: Can we remove or deprecate the field `SliceGatewayRemoteNodeIP` ?

File: api/v1beta1/slicegateway_types.go [hunk 3]
```diff
@@ -53,6 +53,8 @@ type SliceGatewayConfig struct {
...
+	SliceGatewayRemoteNodeIPs []string `json:"sliceGatewayRemoteNodeIps,omitempty"`
 	// Remote Node Port
```
Comment (kmjayadeep) [inline api/v1beta1/slicegateway_types.go:58]: Don't we need multiple node ports?

How about we rename the field to `SliceGatewayRemoteAddresses` and include `ip:port` as the value?

File: api/v1beta1/slicegateway_types.go [hunk 4]
```diff
@@ -73,20 +75,30 @@ type SliceGatewayStatus struct {
...
-	// PodIP is the Ip of the gateway pod running in cluster
+	// Deprecated PodIP is the Ip of the gateway pod running in cluster
 	PodIP string `json:"podIp,omitempty"`
-	// LocalIP is the gateway tunnel ip
+	// PodIPs is the list of Ip of the gateway pods running in cluster
+	PodIPs []string `json:"podIps,omitempty"`
+	//Deprecated LocalIP is the gateway tunnel ip
 	LocalIP string `json:"localIp,omitempty"`
+	// LocalIPs is the gateway tunnel ip
+	LocalIPs []string `json:"localIps,omitempty"`
```
Comment (kmjayadeep) [inline api/v1beta1/slicegateway_types.go:91]: why do we need `LocalIPs` when the same is available in `GwPodInfo` struct?

File: controllers/slicegateway/reconciler.go [hunk 5]
```diff
@@ -59,11 +59,11 @@ type SliceGwReconciler struct {
 	WorkerGWSidecarClient WorkerGWSidecarClientProvider
 	NetOpPods             []NetOpPod
 	EventRecorder         *events.EventRecorder
-	NodeIP                string
+	NodeIP                []string
```
Comment (kmjayadeep) [inline controllers/slicegateway/reconciler.go:62]:
```suggestion
	NodeIPs                []string
```

File: pkg/hub/controllers/slicegateway_controller.go [hunk 6]
```diff
@@ -86,6 +86,7 @@ func (r *SliceGwReconciler) Reconcile(ctx context.Context, req reconcile.Request
 		SliceGatewayRemoteSubnet:    sliceGw.Spec.RemoteGatewayConfig.GatewaySubnet,
 		SliceGatewayHostType:        sliceGw.Spec.GatewayHostType,
 		SliceGatewayRemoteNodeIP:    sliceGw.Spec.RemoteGatewayConfig.NodeIp,
```
Comment (kmjayadeep) [inline pkg/hub/controllers/slicegateway_controller.go:88]: We don't have to reconcile `SliceGatewayRemoteNodeIP` since it is removed right?

**Review by kmjayadeep — CHANGES_REQUESTED** [review]

File: .github/workflows/scripts/start-kind.sh (hunk 1 above)
Comment (kon3m) [reply .github/workflows/scripts/start-kind.sh:187]: I believe there is an issue with the istiosuite

File: api/v1beta1/slicegateway_types.go (hunk 2 above)
Comment (kon3m) [reply api/v1beta1/slicegateway_types.go:54]: removed

**Comment by kon3m** [comment]
Closing this as it is a duplicate of #136

**Pull request closed without merging**